lib/Doctrine/Common/Cache/ArrayCache.php

```diff
@@ -38,11 +38,35 @@ class ArrayCache extends CacheProvider
     private $data = array();
 
     /**
+     * @var array $stats
+     */
+    private $stats = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct()
+    {
+        $this->stats = array(
+            Cache::STATS_HITS => 0,
+            Cache::STATS_MISSES => 0,
+            Cache::STATS_UPTIME => time(),
+            Cache::STATS_MEMORY_USAGE => null,
+            Cache::STATS_MEMORY_AVAILABLE => null,
+        );
+    }
+
+    /**
      * {@inheritdoc}
      */
     protected function doFetch($id)
     {
-        return $this->doContains($id) ? $this->data[$id] : false;
+        if ($this->doContains($id)) {
+            $this->stats[Cache::STATS_HITS]++;
+            return $this->data[$id][0];
+        }
+        $this->stats[Cache::STATS_MISSES]++;
```

Comment:
Use a separate private property for miss stats, instead of storing it in a multidimensional array